cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/PluginInput/index.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import WidgetWrapper from 'components/ConfigurationGroup/WidgetWrapper';
+import * as React from 'react';
+
+const PluginInput = ({
```

Comment:
Seems like one single widget is doing a whole bunch of things. Can we a utility function that return appropriate component instead of one single component handling all? Something along the lines of `AbstractWidgetFactory`

Reply:
I will be sure to make this changes in the separate PR, since I use this component a lot across different PRs!